File: Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/execution/ExpressionResult.java
package org.opencds.cqf.cql.engine.execution;

import java.util.Set;

public class ExpressionResult {
    protected Object value;
    protected Set<Object> evaluatedResources;

    public ExpressionResult(Object value, Set<Object> evaluatedResources) {
        this.value = value;
        this.evaluatedResources = evaluatedResources;
    }

    public Object value() {
        return value;
    }

    public Set<Object> evaluatedResources() {
        return  this.evaluatedResources;
    }
}

File: Src/java/engine/src/main/kotlin/org/opencds/cqf/cql/engine/execution/ExpressionResult.kt
package org.opencds.cqf.cql.engine.execution

class ExpressionResult(
    protected var value: Any?,
    protected var evaluatedResources: MutableSet<Any?>?,
) {
    fun value(): Any? {
        return value
    }

    fun evaluatedResources(): MutableSet<Any?>? {
        return this.evaluatedResources
    }
}
